wolfssl/wolfcrypt/settings.h

```diff
@@ -52,6 +52,12 @@
     extern "C" {
 #endif
 
+/* pick up compiler def; may need to turn on HAVE_PTHREAD for some configs */
+#ifdef _POSIX_THREADS
+    #undef HAVE_PTHREAD
```

Comment:
This looks good to me. Now that this is in settings.h I think it best to change the `_POSIX_THREADS` elsewhere to use `HAVE_PTHREAD`. I also think you should add ` && !defined(SINGLE_THREADED)` and remove it from the refactor you did.

Reply:
cool. I've applied your suggested changes and changed the PR description & title to reflect the global nature of this change